app/src/behaviors/behavior_hold_tap.c

```diff
@@ -645,6 +683,13 @@ static int keycode_state_changed_listener(const zmk_event_t *eh) {
         return ZMK_EV_EVENT_BUBBLE;
     }
 
+    if (undecided_hold_tap->first_press) {
```

Comment:
The if checks can be combined here as well, since `first_press` isn't used unless `hold_while_undecided` is set isn't it?